Geometry_Engine/Query/HashArray.cs

```diff
@@ -66,100 +86,173 @@ public static double[] IHashArray(this IGeometry igeometry)
         [Description("The geometry hash of a Curve is obtained by first retrieving any Sub-part of the curve, if present." +
             "The ISubParts() methods is able to return the 'primitive' curves that a curve is composed of. " +
             "The GeometryHashes are then calculated for the individual parts and concatenated.")]
-        private static double[] HashArray(this ICurve curve, double translationFactor)
+        private static double[] HashArray(this ICurve curve, double translationFactor, BaseComparisonConfig comparisonConfig = null, string fullName = null)
         {
+            if (comparisonConfig?.TypeExceptions?.Any(t => typeof(ICurve).IsAssignableFrom(t)) ?? false)
+                return default;
+
+            if (curve == null)
+                return default;
+
             List<ICurve> subParts = curve.ISubParts().ToList();
 
+            if (!subParts.Any())
+                if (curve is Polyline)
+                    return HashArray(curve.IControlPoints(),
+                        translationFactor,
+                        comparisonConfig: comparisonConfig,
+                        fullName: fullName.AppendPropertyName($"{nameof(ControlPoints)}"));
+                else
+                    return HashArray(curve as dynamic,
+                        translationFactor,
+                        skipEndPoint: false,
+                        comparisonConfig: comparisonConfig,
+                        fullName: fullName);
+
             List<double> hashes = new List<double>();
 
             //Add hash ignoring endpoint for all but last curve
             for (int i = 0; i < subParts.Count - 1; i++)
             {
-                hashes.AddRange(HashArray(subParts[i] as dynamic, translationFactor, true));
+                hashes.AddRange(HashArray(subParts[i] as dynamic,
+                    translationFactor,
+                    skipEndPoint: true,
+                    comparisonConfig: comparisonConfig,
+                    fullName: fullName.AppendPropertyName($"[{i}]"))
+                );
             }
-            //Include endpoint for hasing for last curve
-            hashes.AddRange(HashArray(subParts.Last() as dynamic, translationFactor, false));
+
+            //Include endpoint for hashing for last curve
+            hashes.AddRange(HashArray(subParts.Last() as dynamic,
```

Comment:
What about the cases of disjoint `PolyCurves`? They theoretically are valid.

Reply:
After conversation with @pawelbaran we've agreed this is a thing to tackle separately from this PR, as this requires improving the mechanism that was originally designed for GeometryHash/ToHashArray. Raised issue here: https://github.com/BHoM/BHoM_Engine/issues/3272